client/src/components/LoginPage/index.js

```diff
@@ -1,85 +1,105 @@
 import React, { useState } from "react";
-import { Button, makeStyles } from "@material-ui/core";
-import TextField from "@material-ui/core/TextField";
+import { Button } from "@material-ui/core";
+import { sha256 } from "js-sha256";
+import UserContext from "../../UserContext";
 import Footer from "./Footer";
+import InputField from "./InputField";
 import "./Login.css";
+import Dialog from "../shared/Dialog";
+import axios from "axios";
 
-const styles = makeStyles({
-  textBox: {
-    margin: 10,
-    width: "80%",
-  },
-});
-
-// const [state name, function to update state]
-// body = default state
 const LoginPage = (props) => {
-  const [user, setTempUser] = useState({
+  const { user, setUser } = React.useContext(UserContext);
+  const [emailError, setEmailError] = useState(false);
+  const [passwordError, setPasswordError] = useState(false);
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [tempUser, setTempUser] = useState({
     username: "",
     email: "",
     password: "",
   });
 
   // called when user types into fields
-  const handleChange = (e) => {
+  // if any fields are red (error), remove them at this time
+  // update temp user fields based on which field was typed into
+  const handleChange = (e, tempUser) => {
     const { name, value } = e.target;
-    if (name === "username") {
-      setTempUser({
-        username: value,
-        email: user["email"],
-        password: user["password"],
-      });
-    } else if (name === "email") {
-      setTempUser({
-        username: user["username"],
-        email: value,
-        password: user["password"],
-      });
+    if (name === "email") {
+      setTempUser({ ...tempUser, email: value });
```

Comment:
I'm gonna need you to explain to me how this works because I was referencing your work when doing registration page and really liked how you did this but it simply would not work for me

Reply:
I'll DM you about it